storagemarket/deal_execution.go

```diff
@@ -103,33 +113,39 @@ func (p *Provider) transferAndVerify(ctx context.Context, pub event.Emitter, dea
 
 	// TODO: need to ensure that passing the padded piece size here makes
 	// sense to the transport layer which will receive the raw unpadded bytes
-	execParams := datatransfer.ExecuteParams{
-		TransferType:   deal.TransferType,
-		TransferParams: deal.TransferParams,
-		DealUuid:       deal.DealUuid,
-		FilePath:       deal.InboundFilePath,
-		Size:           uint64(deal.ClientDealProposal.Proposal.PieceSize),
-	}
-	sub, err := p.Transport.Execute(tctx, execParams)
+	handler, err := p.Transport.Execute(tctx, deal.TransferParams, &transporttypes.TransportDealInfo{
+		OutputFile: deal.InboundFilePath,
+		DealUuid:   deal.DealUuid,
+		// TODO Is this the correct size of the actual data without padding ?
+		// I think it's best to explicitly add a "deal size" param to the deal negotiation.
+		DealSize: int64(deal.ClientDealProposal.Proposal.PieceSize),
+	})
 	if err != nil {
 		return fmt.Errorf("failed data transfer: %w", err)
 	}
+	defer handler.Close()
```

Comment:
nit: suggest refactoring this loop into a function so you can call `defer handler.Close()` at the beginning of the function. At the moment it's a little tricky to follow why handler.Close() is called in some places and not others (eg on line 143, but not on line 137)